db/changes.go

```diff
@@ -820,3 +820,106 @@ func (db *Database) getLateFeed(feedHandler *lateSequenceFeed) (<-chan *ChangeEn
 func (db *Database) closeLateFeed(feedHandler *lateSequenceFeed) {
 	db.changeCache.getChannelCache(feedHandler.channelName).ReleaseLateSequenceClient(feedHandler.lastSequence)
 }
+
+/*
+ * Generate the changes for a specific list of doc ID's, only documents accesible to the user will generate
+ * results.  Only supports non-continuous changes, closes buffered channel before returning.
+ */
+func (db *Database) DocIDChangesFeed(userChannels base.Set, explicitDocIds []string, options ChangesOptions) (<-chan *ChangeEntry, error) {
+
+	// Subroutine that creates a response row for a document:
+	output := make(chan *ChangeEntry, len(explicitDocIds))
+	rowMap := make(map[uint64]*ChangeEntry)
+
+	createChangeEntry := func(docid string) *ChangeEntry {
```

Comment:
I (subjectively) feel like this function is getting a bit too big to inline as a function closure. 

What about splitting it out like [this](https://github.com/couchbase/sync_gateway/commit/1da09c6a856a5ea408c3296664ab5cf8f2fed3f6)?  

The potential benefits I see are:

1. Easier to grok the overall flow of `DocIDChangesFeed()`
1. Easier to test the inline function in isolation
1. Easier to refactor the inline function

Reply:
In this case I was trying not to mess with the existing code that much, so didn't consider doing this refactor.  I'd probably rather leave to 2.1.0 to minimize the risk (could revisit in the context of supporting continuous changes w/ DocID filtering).

Reply:
Ok.  How about we file an issue (if one doesn't already exist) or at least putting some todo's in the code?  Now that I'm looking more closely, looks like a lot of this code is duplicated from [existing code](https://gist.github.com/tleyden/88e29b647ddb126b437ce0c7f8082779), which I didn't see initially when reading the diff of the PR.

Reply:
Added a TODO for refactoring this out.